app/javascript/App/components/WatchButton.js

```diff
@@ -3,52 +3,54 @@ import API from '../lib/localAPI'
 import Icon from './Icon'
 
 export default class WatchButton extends Component {
-  state = { coin: null, watching: false }
+  state = { watching: false }
   componentDidMount() {
     const watching = !!this.props.watching
     this.setState({ watching })
   }
+  userIsLoggedIn = !!(this.props.user || this.props.loggedIn)
   handleClick = () => {
-    const { coinID: id, onWatch, onChange } = this.props
-    const { watching } = this.state
-    if (watching) {
-      API.delete(`/watchlist/coins/${id}.json`).then(({ type }) => {
-        if (type === 'success') {
-          this.setState({ coin: null, watching: false })
-          if (onChange) onChange({ id, watching: false })
-        }
-      })
-    } else {
-      API.post('/watchlist/coins.json', { id }).then(
-        ({ type, payload: coin }) => {
-          if (type === 'success') {
-            this.setState({ coin, watching: true })
-            if (onWatch) onWatch()
-            if (onChange) onChange({ id, watching: true })
-          }
-        }
-      )
-    }
+    if (!this.userIsLoggedIn) return
+    /* This component does its own API request & state management since it's
+    sometimes used on its own, and doesn't have access to Redux */
+    const { coinID: id, onWatch, onChange, setUser } = this.props
+    let { watching } = this.state
+    let params = { watchCoin: id }
+    if (watching) params = { unwatchCoin: id }
+    API.patch('/user', params).then(({ type, payload }) => {
+      console.log(params)
+      console.log(payload)
+      if (type === 'success') {
+        if (setUser) setUser(payload)
+        watching = payload.coin_ids.includes(id)
+        this.setState({ watching })
+        if (watching && onWatch) onWatch()
+        if (onChange) onChange(watching)
+      }
+    })
   }
   render() {
     const { watching } = this.state
     let btnClass = 'btn btn-xs'
     if (watching) btnClass += ' btn-blue'
     if (!watching) btnClass += ' btn-gray'
     return (
-      <button onClick={this.handleClick} className={btnClass}>
-        {watching ? (
-          <span>
-            <Icon name="star" solid className="mr1" />
-            Watching
-          </span>
-        ) : (
-          <span>
-            <Icon name="star" regular className="mr1" />
-            Watch
-          </span>
-        )}
-      </button>
+      <div className="dib tooltipped">
+        {!this.userIsLoggedIn && <div className="tooltip">Login to watch</div>}
+        <button onClick={this.handleClick} className={btnClass}>
+          {watching ? (
+            <span>
+              <Icon name="star" solid className="mr1" />
+              Watching
```

Comment:
let me know how you feel about creating a localization file and keeping our strings there. it's very possible we might want to support other languages and it makes the code cleaner. something feels wrong hardcoding strings in components! i just did it in another PR, and i feel like a 🤠

Reply:
Yeah definitely onboard